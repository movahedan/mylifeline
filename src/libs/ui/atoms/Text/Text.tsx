import { classNames } from "@mylifeline/utilities";

import type { CSSProperties, ReactNode } from "react";

type TextProps = {
  as?: keyof HTMLElementTagNameMap;
  inline?: boolean;
  dataTestId?: string;
  style?: CSSProperties;
  className?: string;
  children: ReactNode;
};

export const Text = ({
  as: As = "p",
  inline = false,
  dataTestId,
  style,
  className,
  children,
}: TextProps) => (
  <As
    data-testid={dataTestId}
    style={style}
    className={classNames([
      "prose",
      inline ? "inline-flex items-center" : "flex items-center",
      className || "",
    ])}
  >
    {children}
  </As>
);
