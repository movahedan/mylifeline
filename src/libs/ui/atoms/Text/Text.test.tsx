import { render } from "@testing-library/react";

import { Text } from "./Text";

describe("<Text>", () => {
  it("should match is's snapshot", () => {
    const { container, rerender } = render(<Text>Hello</Text>);
    expect(container).toMatchSnapshot("default");

    rerender(<Text inline>Hello</Text>);
    expect(container).toMatchSnapshot("inline");
  });
});
