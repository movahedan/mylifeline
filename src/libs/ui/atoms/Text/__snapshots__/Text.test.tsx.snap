// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> should match is's snapshot: default 1`] = `
<div>
  <p
    class="prose flex items-center"
  >
    Hello
  </p>
</div>
`;

exports[`<Text> should match is's snapshot: inline 1`] = `
<div>
  <p
    class="prose inline-flex items-center"
  >
    Hello
  </p>
</div>
`;
